src/components/TeacherViews/Onboarding/RoleSelection/RoleSelection.view.tsx

```diff
@@ -61,17 +63,6 @@ const RoleSelectionView = () => {
           formState.errors.role === undefined &&
           formState.errors.other === undefined
         }
-        onSubmit={() => {
-          if (getValues().role === "Other" && !getValues().other) {
-            setError("other", {
-              message: "Please tell us what your role is",
-            });
-          } else if (!getValues().role) {
-            setError("role", {
-              message: "Please select what describes you best",
-            });
-          }
-        }}
```

Comment:
This started to break when I added `noValidate` to the form — I needed this to disable the default browser validation which was being triggered by me adding the `required` attribute to a few inputs.

It seemed to cause a race between the `onSubmit` and `useForm` validation where the validation logic here was being immediately replaced by the rules defined in the Zod schema. This only appeared in tests which i'm putting down to the synthetic nature of the event system/some quirk I don't really understand.

Anyway moving this into a `refine` on the Zod schema gives us the same behaviour avoiding the issue entirely.